### PR TITLE
Reveal header on swipe down (closes #26)

### LINKED ISSUE #26: Add HomeView header

### PR DESCRIPTION
**Pull request by wilsonpage**

### CODE REVIEW

**Comment by wilsonpage** [comment]
@gmarty this PR is more complex, but implements the UX in a more conventional way.

There is a distinct separation between `ListView` and header. The header snaps to either open/closed, so is never shown partially open. We also have the ability to apply subtle reveal effects tied to the show/hide gesture.

This took me longer than expected, but I've learnt a lot about how to work with the `PanResponder`.

File: js/scenes/Home.js [hunk 1]
```diff
@@ -32,10 +34,333 @@ export class Home extends Component {
...
+
+      // apply the delta to the offset ensuring that when the
+      // user's finger drags beyond the clamped zone, the moment
+      // they drag it in the opposite direction, the view responds
+      this.setOffset(this.yOffset - deltaY);
+    });
+  }
+
+  componentWillMount() {
+    this._panResponder = PanResponder.create({
```
Comment (gmarty) [inline js/scenes/Home.js]: In the POI Map PR you moved this to its own methods which I think makes it clearer.

File: js/scenes/Home.js [hunk 2]
```diff
@@ -32,10 +34,333 @@ export class Home extends Component {
...
+  isPanningDown() {
+    return this.baseY > this.previousBaseY;
+  }
+
+  /**
+   * Called when the `ListView` scrolls.
+   *
+   * Gives us an opportunity to store the
+   * last known scroll position and close
+   * the header shld it happen to be open.
```
Comment (gmarty) [inline js/scenes/Home.js]: nit: I'd avoid abbreviations in the comments.

File: js/scenes/Home.js [hunk 3]
```diff
@@ -32,10 +34,333 @@ export class Home extends Component {
...
+      toValue,
+      velocity: 20,
+      overshootClamping: true,
+      tension: 100,
+      easing: Easing.inOut(Easing.quad),
+    }).start();
+  }
+
+  /**
+   * Clamps the
```
Comment (gmarty) [inline js/scenes/Home.js]: Comment unterminated?

File: js/scenes/Home.js [hunk 4]
```diff
@@ -32,10 +34,333 @@ export class Home extends Component {
...
+
+        this.setScrollEnabled(scrollEnabled);
+
+        return shouldSetPanHandler;
+      },
+
+      /**
+       * Always attempt to keep control of the gesture.
+       *
+       * @return {Booelan}
```
Comment (gmarty) [inline js/scenes/Home.js]: nit: Typo in `Boolean`.

**Review by gmarty — APPROVED** [review]
The code and the effect are beautiful!

Can you make the status bar transparent (slightly opaque in Android) and have the content to be shown underneath? This is going to make the app much more beautiful than showing a dark grey status bar.

Once again I'm not super familiar with PanResponder so I reviewed this code to the best of my ability. Thankfully it's largely commenting so that will make maintenance easier.

That would be nice to take the time to explain your solution to the rest of the team so we all understand how it works.

File: js/scenes/Home.js (hunk 2 above)
Comment (wilsonpage) [reply js/scenes/Home.js]: Oops typo

**Comment by wilsonpage** [comment]
@gmarty I've added the scroll under status-bar comment as a follow-up issue #43.

**Pull request merged**